File: frontend/src/types/order.ts
export type OrderType = "dine-in" | "pickup";

export enum OrderStatus {
  New = "New",
  InProgress = "InProgress",
  Ready = "Ready",
  Delivered = "Delivered",
  Cancelled = "Cancelled",
}

export interface OrderItem {
  id: string;
  name: string;
  quantity: number;
  unitPrice: number;
}

export interface Order {
  id: string;
  tenantId?: string;
  orderType: OrderType;
  tableNumber?: number;
  status: OrderStatus;
  items: OrderItem[];
  total: number;
  notes?: string;
  createdAt: string;
  updatedAt: string;
}

File: backend/src/DineOS.Domain/Entities/Order.cs
using DineOS.Domain.Common;
using DineOS.Domain.Enums;

namespace DineOS.Domain.Entities;

public class Order : TenantAuditingEntity
{
    public string OrderType { get; set; } = string.Empty;
    public int? TableNumber { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.New;
    public decimal Total { get; set; }
    public string? Notes { get; set; }
    public ICollection<OrderItem> Items { get; set; } = [];
}
